query_optimizer/PhysicalGenerator.cpp

```diff
@@ -176,6 +177,8 @@ P::PhysicalPtr PhysicalGenerator::optimizePlan() {
     rules.emplace_back(new AttachLIPFilters());
   }
 
+  rules.push_back(std::make_unique<Partition>(optimizer_context_));
```

Comment:
The other question is that -- is this rule always needed to be applied (e.g. even in single-node mode)? If not, we may have a flag or `#ifdef` to conditionally apply this rule.